app/default/routes.py

```diff
@@ -74,18 +78,87 @@ def not_eligible():
 
 @default_bp.route("/tasklist/<application_id>", methods=["GET"])
 def tasklist(application_id):
-    return tasklist_page(application_id)
+    """
+    Returns a Flask function which constructs a tasklist for an application id.
+
+    Args:
+        application_id (str): the id of an application in the application store
+
+    Returns:
+        function: a function which renders the tasklist template.
+    """
+    application_response = get_application_data(application_id)
+    application = Application.from_dict(application_response)
+    if not (application and application.sections):
+        return render_template(
+            "application_unknown.html", application_id=application_id
+        )
```

Comment:
I guess the sections error should never happen by a user's action, so I would suggest throwing a 500 in that case and an exception, if the application's not found, fine to return a 404 response.

Reply:
agreed – removed unnecessary page and sending to a 404 if application not found 👍